### ISSUE TITLE
Unable to get oauth2 token (PYTHON-MSAL library)

### ISSUE THREAD

**Issue by victortorrecilla** [Azure/azure-cli#23370]
HI,

I'm trying to get the uth token for handling my application in azure. For doing that I'm using the python library python and Im follwowing this tutorial on the documentation of Microsoft https://docs.microsoft.com/en-us/python/api/overview/azure/active-directory?view=azure-python

I'm trying  to connect but everytime that I try to connect i'm getting the same error

requests.exceptions.ConnectionError: HTTPSConnectionPool(host='login.microsoftonline.com', port=443): Max retries exceeded with url: /aa06dce7-99d7-403b-8a08-0c5f50471e64/v2.0/.well-known/openid-configuration (Caused by ProtocolError('Connection aborted.', OSError(0, 'Error')))

is someone else having this issue?

thanks,

**Comment by yonzhan** [comment]
@rayluo to help with

**Comment by rayluo** [comment]
Hi Victor, can your machine connect to that URL https://login.microsoftonline.com/aa06dce7-99d7-403b-8a08-0c5f50471e64/v2.0/.well-known/openid-configuration ? It works for me. Also you need to make sure your machine does NOT somehow map `login.microsoftonline.com` into a different IP.

**Comment by victortorrecilla** [comment]
Hi Ray,

thanks a lot for anwering me. I've checked the URL and I'm able to reach it.

about this part _"Also you need to make sure your machine does NOT somehow map login.microsoftonline.com into a different IP."_ I'm not sure if I understand you completly, how can I check if the login.microsoftonline.com is mapped into a different IP?

thanks,

**Comment by rayluo** [comment]
OK, let me rephrase.

Now that both you and I are able to visit that URL, this means the URL looks right. But that URL caused a ConnectionError from the machine that your python script was running on. So, at this point, you (and probably only you) can investigate why that URL was unreachable from your machine (which might not be your desktop). Was it a VM with limited outbound link? Was it a container? We just don't know.

**Comment by victortorrecilla** [comment]
Hi Ray,

I've seen the problem, I've resolved the proxy and with this everything is working, however, the code is failing in the following point:

"result = app.acquire_token_by_one_of_the_actual_method(..., scopes=["User.Read"])
AttributeError: 'PublicClientApplication' object has no attribute 'acquire_token_by_one_of_the_actual_method'"

thanks,

**Comment by victortorrecilla** [comment]
the problem was the proxy... I've added the proxy and everything is woking, however, I'm trying to get the token thru acquire_token_by_auth_code_flow but there is a parameter that I don't know where I can get the value "auth_response (dict) – A dict of the query string received from auth server."

thanks,

**Comment by rayluo** [comment]
Looks like you are trying to use MSAL Python and develop your own script, rather than using Azure CLI? If so, I would suggest you close this issue in Azure CLI repo, and create your generic MSAL Python question on Stackoverflow. The community will help you there.

**Comment by victortorrecilla** [comment]
Hi Ray,

thanks a lot for all your. Well I'm trying to connect to an api and I've seen this method where I don't know what means the parameter that I've sent to you. 

I've tried the rest of methods for getting the token but I'm always receiving the following message "AADSTS65002: Consent between first party application 'XXXXXXXXXXXX' and first party resource '00000003-0000-0000-c000-000000000000' must be configured via preauthorization - applications owned and operated by Microsoft must get approval from the API owner before requesting tokens for that API." this is something that I'm checking with the admin of the app,

but well as you mention, if you think  is better to get help int stackverflow I will try it in this way.

again thanks a lot for all your help.

cheers

**Comment by jiasli** [comment]
@rayluo is right. If this issue is not regarding Azure CLI, please create your issue at https://github.com/AzureAD/microsoft-authentication-library-for-python
